File: app/src/main/java/mx/leo/sample/easyrecycler/adapter/SampleItemAdapter.java
package mx.leo.sample.easyrecycler.adapter;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import mx.leo.easyrecycler.adapter.EasyAdapter;
import mx.leo.sample.easyrecycler.R;
import mx.leo.sample.easyrecycler.model.Language;

/**
 * Created by Braulio on 12/12/2016.
 */
public class SampleItemAdapter extends EasyAdapter<SampleItemViewHolder, Language> {

    @NotNull @Override public SampleItemViewHolder createHolder(@Nullable ViewGroup parent, int viewType) {
        return new SampleItemViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.sample_item_layout,parent,false));
    }

    @Override public void onBindItemViewHolder(@NotNull SampleItemViewHolder holder, Language language, int position) {
        holder.bindItem(language);
    }
}

File: samplekotlin/src/main/java/brauliomendez/com/kotlin/adapter/SampleItemAdapter.kt
package brauliomendez.com.kotlin.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import brauliomendez.com.kotlin.R
import brauliomendez.com.kotlin.model.Language
import brauliomendez.com.kotlin.viewholder.SampleItemViewHolder
import mx.leo.easyrecycler.adapter.EasyAdapter

/**
 * Created by Braulio on 26/12/2016.
 */
class SampleItemAdapter : EasyAdapter<SampleItemViewHolder, Language>() {

    override fun createHolder(parent: ViewGroup?, viewType: Int): SampleItemViewHolder {
        return SampleItemViewHolder(LayoutInflater.from(parent?.context).inflate(R.layout.item_layout, parent, false))
    }

    override fun onBindItemViewHolder(holder: SampleItemViewHolder, item: Language, position: Int) {
        holder.bindItem(item)
    }
}